File: src/main/java/factory/creator/UltraVehicleFactory.java
package factory.creator;

import factory.product.UltraBus;
import factory.product.UltraTruck;
import factory.product.Vehicle;
import factory.product.VehicleType;

public class UltraVehicleFactory extends VehicleFactory {
    @Override
    protected Vehicle createVehicle(VehicleType vehicleType) {
        return switch(vehicleType) {
            case BUS -> new UltraBus();
            case TRUCK -> new UltraTruck();
        };
    }
}

File: kotlin/src/main/kotlin/factory/creator/UltraVehicleFactory.kt
package factory.creator

import factory.product.UltraBus
import factory.product.UltraTruck
import factory.product.Vehicle
import factory.product.VehicleType

class UltraVehicleFactory : VehicleFactory() {
    override fun createVehicle(vehicleType: VehicleType): Vehicle {
        return when (vehicleType) {
            VehicleType.BUS -> UltraBus()
            VehicleType.TRUCK -> UltraTruck()
        }
    }
}